pkg/cmd/repo/sync/sync_test.go

```diff
@@ -0,0 +1,434 @@
+	stubConfirm := func(as *prompt.AskStubber) {
+		as.StubOne(true)
+	}
+
+	tests := []struct {
+		name       string
+		tty        bool
+		opts       *SyncOptions
+		remotes    []*context.Remote
+		httpStubs  func(*httpmock.Registry)
+		gitStubs   func(*syncfakes.FakeGitClient)
+		askStubs   func(*prompt.AskStubber)
+		wantStdout string
+		wantStderr string
+		wantErr    bool
+		errMsg     string
+	}{
+		{
+			name: "sync local repo with parent - tty",
+			tty:  true,
+			opts: &SyncOptions{},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query RepositoryInfo\b`),
+					httpmock.StringResponse(`{"data":{"repository":{"defaultBranchRef":{"name": "trunk"}}}}`))
+			},
+			askStubs:   stubConfirm,
+			wantStdout: "✓ Synced .:trunk from OWNER/REPO:trunk\n",
+		},
+		{
+			name: "sync local repo with parent - notty",
+			tty:  false,
+			opts: &SyncOptions{
+				SkipConfirm: true,
+			},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query RepositoryInfo\b`),
+					httpmock.StringResponse(`{"data":{"repository":{"defaultBranchRef":{"name": "trunk"}}}}`))
+			},
+			askStubs:   stubConfirm,
+			wantStdout: "",
+		},
+		{
+			name: "sync local repo with specified source repo",
+			tty:  true,
+			opts: &SyncOptions{
+				SrcArg: "OWNER2/REPO2",
+			},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query RepositoryInfo\b`),
+					httpmock.StringResponse(`{"data":{"repository":{"defaultBranchRef":{"name": "trunk"}}}}`))
+			},
+			askStubs:   stubConfirm,
+			wantStdout: "✓ Synced .:trunk from OWNER2/REPO2:trunk\n",
+		},
+		{
+			name: "sync local repo with parent and specified branch",
+			tty:  true,
+			opts: &SyncOptions{
+				Branch: "test",
+			},
+			askStubs:   stubConfirm,
+			wantStdout: "✓ Synced .:test from OWNER/REPO:test\n",
+		},
+		{
+			name: "sync local repo with parent and force specified",
+			tty:  true,
+			opts: &SyncOptions{
+				Force: true,
+			},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query RepositoryInfo\b`),
+					httpmock.StringResponse(`{"data":{"repository":{"defaultBranchRef":{"name": "trunk"}}}}`))
+			},
+			gitStubs: func(fgc *syncfakes.FakeGitClient) {
+				fgc.HasLocalBranchReturns(true)
+				fgc.IsAncestorReturns(false, nil)
+			},
+			askStubs:   stubConfirm,
+			wantStderr: "! Using --force will cause diverging commits on .:trunk to be discarded\n",
+			wantStdout: "✓ Synced .:trunk from OWNER/REPO:trunk\n",
+		},
+		{
+			name: "sync local repo with parent and not fast forward merge",
+			tty:  true,
+			opts: &SyncOptions{},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query RepositoryInfo\b`),
+					httpmock.StringResponse(`{"data":{"repository":{"defaultBranchRef":{"name": "trunk"}}}}`))
+			},
+			gitStubs: func(fgc *syncfakes.FakeGitClient) {
+				fgc.HasLocalBranchReturns(true)
```

Comment:
1. Does counterfeiter fail a test if this stubbed method was never invoked?
2. Does counterfeiter fail a test if there were more invocations than stubs for the same method?

These the features I would ideally want from a stubbing solution.

Reply:
1. No
2. No

Counterfeiter does not provide that functionality by default, but gives the user the ability to check those things through the generated interface it creates, so we could write an assertion for the number of times a function was called.